hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/TokenClient.java

```diff
@@ -274,32 +276,56 @@ public NetworkTransactionResponse createNonFungibleToken(
     }
 
     public NetworkTransactionResponse associate(ExpandedAccountId accountId, TokenId token) {
-        TokenAssociateTransaction tokenAssociateTransaction = new TokenAssociateTransaction()
-                .setAccountId(accountId.getAccountId())
-                .setTokenIds((List.of(token)))
-                .setTransactionMemo(getMemo("Associate w token"));
-
-        var keyList = KeyList.of(accountId.getPrivateKey());
-        var response = executeTransactionAndRetrieveReceipt(tokenAssociateTransaction, keyList);
-        log.info("Associated account {} with token {} via {}", accountId, token, response.getTransactionId());
-        tokenAccounts.add(new TokenAccount(token, accountId));
-        return response;
+        var account = new EntityId(accountId);
+        TokenAccountPair key = new TokenAccountPair(token, account);
+
+        synchronized (associations) {
+            associations.computeIfAbsent(key, k -> {
```

Comment:
Since you're using a concurrent hash map synchronized is not necessary. Also, separate call to get() is not needed either:
```suggestion
        return associations.computeIfAbsent(key, k -> {
```

Reply:
Great idea! Now it is updated